Imp

NaokiTsuchiya opened the pull request.

NaokiTsuchiya committed:
Imp

NaokiTsuchiya committed:
Expand README.md with package usage and detailed installation instructions

The README.md file has been expanded to include more comprehensive usage sample and detailed installation instructions. It now provides an in-depth explanation on how to use the `naoki-tsuchiya/ray-di-psr-container` package, along with inserted PHP codes to visualize its implementation. It also covers system-level instructions on how to install and check the package on your local machine. As a developer new to this package, these additional details would be significantly helpful.

NaokiTsuchiya committed:
Refactor Identity Parsing

This commit introduces a dedicated IdentityParser class to handle the logic of parsing identity strings. Previously, this logic was mixed in the Psr11Injector class. The new class separates the concerns better and makes the code easier to understand and manage. To accommodate this change, the references to name separators in Psr11Injector and tests have been updated from Psr11Injector::NAME_SEPARATOR to IdentityParser::NAME_SEPARATOR.

NaokiTsuchiya committed:
Improve code by refactoring and adding new features

This commit includes a refactoring of the code and addition of new features. Specifically, we have marked the `IdentityParser` class as internal for added privacy, created a new class `IdentityStringGenerator` to generate unique identifiers for interface and name combinations, and developed a new test for the `IdentityGenerator` class. We also refactored the `Psr11InjectorTest.php` to utilise the new `IdentityGenerator` class. These changes enhance readability and maintainability of the code.

koriym commented on src/IdentityParser.php:
```diff
@@ -0,0 +1,89 @@
+use function substr;
+
+/** @internal */
+class IdentityParser
```
```suggestion
final class IdentityParser
```

koriym commented on src/Psr11Injector.php:
```diff
@@ -0,0 +1,64 @@
+use Ray\Di\Name;
+use Throwable;
+
+class Psr11Injector implements ContainerInterface, InjectorInterface
```
```suggestion
final class Psr11Injector implements ContainerInterface, InjectorInterface
```

koriym commented on tests/Fake/Attribute/Left.php:
```diff
@@ -0,0 +1,14 @@
+
+#[Attribute]
+#[Qualifier]
+class Left
```
```suggestion
final class Left
```

+他のアトリビュートも

koriym commented:
質問

`Container`と`Injector`は一緒にした方がいいですか？

koriym commented on src/Psr11Injector.php:
```diff
@@ -0,0 +1,64 @@
+            $instance = $this->getInstance(...$parsedId);
+        } catch (RayDiUnbound $e) {
+            throw new Unbound($e->getMessage(), 0, $e);
+        } catch (Throwable $e) { // @codeCoverageIgnoreStart
```
`@codeCoverageIgnoreStart`を外してテストを作成してもいいかと思います。

NaokiTsuchiya replied:
InjectorInterface を実装した Fake クラスを使ったテストを追加しました。

eb9197ca0d08c80500f08fd9ab0c7c955095c2ef

koriym commented:
もしContainerInterfaceだけサポートするなら名前はContainerとかPsrContainerとかですかね。

NaokiTsuchiya committed:
Apply suggestions from code review

Co-authored-by: Akihito Koriyama <akihito.koriyama@gmail.com>

NaokiTsuchiya committed:
Make classes final in test files

Marking classes as `final` makes them explicitly non-extensible, improving code safety and clarity. This change enforces a stricter contract for these test classes, ensuring they are not inadvertently subclassed.

NaokiTsuchiya committed:
Fix location of IdentityStringGeneratorTest.php

Moved the test file to a more generic and accessible location. The change simplifies the directory structure and improves maintainability.

NaokiTsuchiya committed:
Remove InjectorInterface implementation

The `getInstance` method was removed from `Psr11Injector` as it was unnecessary.

NaokiTsuchiya committed:
Rename Psr11Injector to PsrContainer.

This renaming improves clarity and consistency in naming conventions, reflecting a more descriptive and concise identifier. All related references, including class names, file names, test cases, and configuration files, have been updated accordingly.

NaokiTsuchiya committed:
Add test for exception handling in PsrContainer

Introduce a new test to verify ContainerException is thrown when the injector fails. Added a fake injector to simulate exceptions and updated PsrContainer to improve code clarity by removing ignored coverage annotations.

NaokiTsuchiya commented:
InjectorInterface の実装をやめて、`PsrContainer` にクラス名を変更しました。

02f36c7ae14466d9a9c07383596dd3c275a77446
8617629dd5b24c7534dd7452f5634f445e43d7ba
